### DISCUSSION TITLE
The not too difficult way to create .h format smoothfonts from ttf files installed on your computer

### DISCUSSION THREAD

**Discussion by Rollmops67** [Bodmer/TFT_eSPI#2452] [Q&A - Fonts]
Hello from France ! (and sorry for my so so english)
First I have to say a big THANK YOU to Bodmer for TTF_eSPI, the more I use it, the more I see new possibilities, and the more I see the complexity and the work involved in this library.

As a frenchie I was a bit disapointed of the 7 bit coding of fonts (I understand that this is based on GFX history and to maintain compatibility), meaning I cannot print accented characters.
While investigating I saw that it is possible to create smoothfonts with 8 bits meaning that it would be (maybe ?) possible to print accented characters.
And as smoothfonts look way better than normal fonts I was interested to create some of them.

Because I had to search for hours to find the "not too difficult way" to achieve this I thought it could be intersting to share my findings here !

I'm not a beginner in C++, but I've not (yet) a very big knowledge, and the solution proposed by Bodmer here :
`https://github.com/Bodmer/TFT_eSPI/tree/master/Tools/Create_Smooth_Font/Create_font`
was a little too complex for me.
Furthermore Bodmer's sketch says :
// Once created the files must be loaded into the ESP32 or ESP8266 SPIFFS memory
// using the Arduino IDE plugin detailed here:
// https://github.com/esp8266/arduino-esp8266fs-plugin
// https://github.com/me-no-dev/arduino-esp32fs-plugin

This also was a little discouraging for me.

But while searching for a solution I found it on Seeedstudio's internet site, here :
[https://wiki.seeedstudio.com/Wio-Terminal-LCD-Anti-aliased-Fonts/#generate-vlw-font-file](https://wiki.seeedstudio.com/Wio-Terminal-LCD-Anti-aliased-Fonts/#generate-vlw-font-file)
There is the program "Processing" (link on the Seeedstudio page) that you can downlod on your computer (different platforms supported)
So I did, and now I am able to create .vlw files from TTF files.
The interesting part is that in the " Character Selector " window of the software I can choose which characters I will include in my .vlw file, and if I tick the first and second options '"Basic Latin" and "Latin-1 supplement" I can in fact print accented characters. Yeeaahhh !

Now I just had to find a way to transform the new .vlw file in an .h file that I can simply include in my sketch and choose with the "loadFont" command.

Once more the solution was on one of Bodmers page, here :
`https://github.com/Bodmer/TFT_eSPI/blob/master/examples/Smooth%20Fonts/FLASH_Array/Font_Demo_1_Array/NotoSansBold15.h`

It's explained in the first 12 lines and straightforward.
So now we can generate our .h smoothfont files even with accented characters.

Don't forget to declare your smoothfonts (the font file has to be in the sketch folder) at the beginning of your sketches with
**#include "policename.h"**, and then you can simply load the font with **txt.loadFont(policename);** when you need it and unload it after usage with **txt.unloadFont();**, to save memory.

Roland

**Comment by Bodmer** [comment]
Thanks for this.

If you need increased control over the character set selection there is a processing sketch here:
https://github.com/Bodmer/TFT_eSPI/tree/master/Tools/Create_Smooth_Font

Load the [Processing IDE](https://processing.org/) and run the sketch within that Processing IDE (not with the Arduino IDE!). There are detailed instructions in the header notes of the Processing sketch.

Note that the Processing IDE is like the Arduino IDE, you can load Processing sketches and run them on the PC.
Enjoy.
